portfolio/src/main/webapp/materialize.js

```diff
@@ -12412,35 +12412,72 @@ if (dark_mode == "no") {
 
 function createListElement(text) {
   const liElement = document.createElement('li');
-  liElement.innerText = text;
+  const button = document.createElement('button');
+  const icon = document.createElement('i');
+  icon.className = "material-icons";
+  icon.innerText = "cancel";
+  button.className = "btn-small";
+  button.appendChild(icon);
+  button.addEventListener('click', () => {
+      deleteComm(text);
+      location.reload();
+  });
+  liElement.appendChild(button)
+  liElement.insertAdjacentText("beforeend", text);
   return liElement;
 }
-var i;
+
+function deleteComm(text) {
+    const params = new URLSearchParams();
+    params.append('text', text);
+    fetch("/deleteOne", {
+        method: "POST", 
+        body: params
+    });
+}
+
+function createParagraphElem(text) {
+    const pElement = document.createElement('p');
+    pElement.innerText = text;
+    return pElement;
+}
+
+var comment_set;
 var curr_comment = 0;
+var num_comments;
+
 function getComments() {
     fetch('/data').then(response => response.json()).then((comments) => {
         const commentList = document.getElementById('comments');
         commentList.innerHTML = '';
-        for (i = 0; i < 5; i++) {
-            if (comments[i] != undefined) {
-                commentList.appendChild(createListElement(comments[i]))
+        num_comments = 0;
+        comment_set = 0;
+        for (comment_set; comment_set < 5; comment_set++) {
+            if (comments[comment_set] != undefined) {
+                commentList.appendChild(createListElement(comments[comment_set]));
             }
             curr_comment++;
         }
+        while (comments[num_comments] != undefined) {
```

Comment:
can we use comments.length instead of `while`

Reply:
Done.